### ISSUE TITLE
Call to a member function isFuture () on string

### ISSUE THREAD

**Issue by adrianoGaspar** [laravel/cashier#532]
After updating to version 7.1.1 (Updating laravel / cashier (v7.1.0 => v7.1.1): Checking out 5e2fa78a5b) the following error occurred: Call to a member function isFuture () on string.
In
/vendor/laravel/cashier/src/Billable.php
/ **
      * Determine if the Stripe model is a "generic" trial at the model level.
      *
      * @return bool
      * /
     public function onGenericTrial ()
     {
         return $ this-> trial_ends_at && $ this-> trial_ends_at-> isFuture ();
     }

This is a bug?

I am using exactly as shown in the laravel manual:

if ($user->onGenericTrial()) { // User is within their "generic" trial period... }

it was 100% and after this update the error occurred, I had to go back to the previous version. Did no one pass this or can tell me what to do?

**Comment by adrianoGaspar** [comment]
This is a bug?

**Comment by UXandre** [comment]
@adrianoGaspar Adiraino Did you manage to solve this? We have got the same problem after composer update. would appreciate it if you could share your thoughts?

**Comment by adrianoGaspar** [comment]
no, I went back to the previous version ("laravel/cashier": "7.1.0") and am waiting for some response, I posted this problem minutes after the update but so far nothing.

**Comment by UXandre** [comment]
@adrianoGaspar Thanks for getting back to me. It's interesting as it actually works on my colleague's laptop but not mine. He is using Cashier 7.1 though.

**Comment by UXandre** [comment]
@adrianoGaspar Hi Adriano, my talented colleague just sorted out the issue. In short, in the latest version you will need to [be sure to add a date mutator for trial_ends_at to your model definition.](https://laravel.com/docs/5.6/billing#subscription-trials). 

Like this:

<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class User extends Model
{
    /**
     * The attributes that should be mutated to dates.
     *
     * @var array
     */
    protected $dates = [
        'created_at',
        'updated_at',
        'deleted_at'
    ];
}

After that it should all work.

**Comment by adrianoGaspar** [comment]
As always the details. Thank you for sharing. Works perfectly now.

**Comment by ricardov03** [comment]
Hello everyone! Thanks so much for this particular issue conversation. Special thanks to @UXandre for finding the needle in the haystack. I found another inside related to the same issue. If you use `protected $casts` with one of the date fields, it will be returning the same isFuture() error. So be sure to only use `protected $date`.

**Comment by vwmoose** [comment]
Just updated from a previous version of Cashier and I'm now getting this same error. 
`Call to a member function isFuture() on string`

Have tried the above and added trial_ends_at to the $casts without success.  Anyone else have any ideas or the solution?

**Comment by vwmoose** [comment]
For anyone who ends up here with the same issue.  I tracked it down to another use of `isFuture()` in the Subscription.php class line 397 in onGracePeriod() where ends_at is tested.  For some reason the casting of `ends_at` as a datetime within Subscription.php is not working for me, maybe because I've created my own Subscription.php and extended the package instance within my project.  Added the casting to my class resolved the issue.  Hopefully it might help someone else. ;)
